wire/msgtx.go

```diff
@@ -461,6 +582,25 @@ func (msg *MsgTx) BtcDecode(r io.Reader, pver uint32, enc MessageEncoding) error
 		msg.TxIn[i].SignatureScript = scripts[offset:end:end]
 		offset += scriptSize
 
+		for j := 0; j < len(msg.TxIn[i].Witness); j++ {
```

Comment:
Good point, move the loop to right after the signature script buffers have been freed up.